### PR TITLE
Document inclusive ranges.

### PR DESCRIPTION
**Pull request by alercah**
Provides the reference documentation for rust-lang/rust#28237.

### CODE REVIEW

File: src/expressions/match-expr.md [hunk 1]
```diff
@@ -95,18 +95,23 @@ match x {
...
-    2 ... 9 => "a few",
+    2 ..= 9 => "a few",
     _      => "lots"
 };
+
+assert_eq!(message, "a few");
 ```
 
+Other forms of [range] \(`..` for an exclusive range, or any range with one or
+both endpoints left unspecified) are not currently supported in matches.
```
Comment (matthewjasper) [inline src/expressions/match-expr.md]: I don't think that 'currently' is needed, the reference only talks about Rust at this point in time.

Comment (alercah) [reply src/expressions/match-expr.md]: Done. I also remarked that `...` is still accepted for backwards compatibility.

**Comment by alercah** [comment]
According to rust-lang/rust#28237, all that's left is docs, so hopefully once all my doc PRs are good it will be ready to go.

**Comment by matthewjasper** [comment]
`..=` is stable now, can you fix the merge conflict so that this can be merged?

**Comment by alercah** [comment]
Done.

**Comment by steveklabnik** [comment]
This just landed as stable in nightly. Let's restart travis!

**Comment by steveklabnik** [comment]
🎊

**Pull request merged**